crates/bevy_sprite/src/render/mod.rs

```diff
@@ -622,154 +633,159 @@ pub fn prepare_sprites(
-                            });
-                        }
-                        colored_index += QUAD_INDICES.len() as u32;
-                        existing_batch.unwrap().1.range.end = colored_index;
-                    } else {
-                        for i in QUAD_INDICES {
-                            sprite_meta.vertices.push(SpriteVertex {
-                                position: positions[i],
-                                uv: uvs[i].into(),
-                            });
-                        }
-                        index += QUAD_INDICES.len() as u32;
-                        existing_batch.unwrap().1.range.end = index;
-                    }
-                    transparent_phase.items[batch_item_index].batch_size += 1;
-                } else {
-                    batch_image_handle = HandleId::Id(Uuid::nil(), u64::MAX);
+                // Override the size if a custom one is specified
+                if let Some(custom_size) = extracted_sprite.custom_size {
+                    quad_size = custom_size;
                 }
+                let transform = extracted_sprite.transform.affine()
+                    * Affine3A::from_scale_rotation_translation(
+                        quad_size.extend(1.0),
+                        Quat::IDENTITY,
+                        (quad_size * (-extracted_sprite.anchor - Vec2::splat(0.5))).extend(0.0),
+                    );
+
+                // Store the vertex data and add the item to the render phase
+                sprite_meta
+                    .sprite_instance_buffer
+                    .push(SpriteInstance::from(
+                        &transform,
+                        &extracted_sprite.color,
+                        &uv_offset_scale,
+                    ));
+
+                if batch_image_changed {
+                    batch_item_index = item_index;
+
+                    batches.push((
+                        item.entity,
+                        SpriteBatch {
+                            image_handle_id: batch_image_handle,
+                            range: index..index,
+                        },
+                    ));
+                }
+
+                transparent_phase.items[batch_item_index].batch_size += 1;
+                batches.last_mut().unwrap().1.range.end += 1;
+                index += 1;
             }
         }
         sprite_meta
-            .vertices
-            .write_buffer(&render_device, &render_queue);
-        sprite_meta
-            .colored_vertices
+            .sprite_instance_buffer
             .write_buffer(&render_device, &render_queue);
+
+        if sprite_meta.sprite_index_buffer.len() != 6 {
+            sprite_meta.sprite_index_buffer.clear();
+
+            // NOTE: This code is creating 6 indices pointing to 4 vertices.
+            // The vertices form the corners of a quad based on their two least significant bits.
+            // 01   11
+            //
+            // 00   01
```

Comment:
Shouldn't the first one be `10`? 

```suggestion
            // 10   11
            //
            // 00   01
```

Reply:
Yes, good catch.